option.go

```diff
@@ -325,13 +330,75 @@ func (optSet *OptionSet) SetDefaults() error {
 			)
 			continue
 		}
-		(*optSet)[i].ValueSource = ValueSourceDefault
-		if err := opt.Value.Set(opt.Default); err != nil {
+		groupByValue[opt.Value] = append(groupByValue[opt.Value], opt)
+	}
+
+	sortOptionByValueSourcePriorityOrDefault := func(a, b *Option) int {
+		if a.ValueSource != b.ValueSource {
+			for _, vs := range valueSourcePriority {
+				if a.ValueSource == vs {
+					return -1
+				}
+				if b.ValueSource == vs {
+					return 1
+				}
+			}
+		}
```

Comment:
This is totally fine, and an optimized solution. My morning brain just took a second to understand it.

Can you just throw a comment like, `// Sorts by value source then a default value being set` 

---

I like using subtraction like this for compare functions.

```
if a.ValueSource != b.ValueSource {
    return slices.Index(valueSourcePriority, a.ValueSource) - slices.Index(valueSourcePriority, b.ValueSource)
}
```

Reply:
That's clean, thanks! 👍🏻